src/compiler.ml

```diff
@@ -460,7 +460,7 @@ and block =
   | Clauses of (preterm,Ast.Structured.attribute) Ast.Clause.t list (* TODO: use a map : predicate -> clause list to speed up insertion *)
   | Namespace of string * pbody
   | Shorten of C.t Ast.Structured.shorthand list * pbody
-  | Constraints of constant list * prechr_rule list * pbody
+  | Constraints of constant list * constant list * prechr_rule list * pbody
```

Comment:
```suggestion
  | Constraints of constraints_block * pbody
 and constraints_block = {
   clique : constant list;
   ctx_filter : constant list;
   rules : prechr_rule list;
 }
```

and use the type also below instead of the triple